File: back/src/main/java/com/openclassrooms/mddapi/models/Article.java
package com.openclassrooms.mddapi.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SourceType;

import javax.persistence.*;
import java.time.Instant;
@Data
@NoArgsConstructor
@Entity
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String content;
    @ManyToOne
    @JoinColumn(name="owner_id",nullable = false)
    private User user;
    @ManyToOne
    @JoinColumn(name="theme_id", nullable = false)
    private Theme theme;

    @CreationTimestamp
    private Instant created_At;


}

File: front/src/app/common/models/article.ts
import { Comment } from "./comment";
import { Topic } from "./topic";
import { UserResponse } from "./userResponse";
export interface Article {

  id: number;
  title: string;
  content: string;
  user: UserResponse;
  topicDTO: Topic;
  comments: Comment[];
  createdAt: string; 
  }